### ISSUE TITLE
Custom option in label property of Text Symbolizer freezes the application

### ISSUE THREAD

**Issue by allyoucanmap** [geosolutions-it/MapStore2#6951]
## Description

If a user try to add a custom option in the label symbolizer the application freezes

## How to reproduce

- login as admin
- open a map with a vector layer with style editing permission
- select or add a new style
- add a text symbolizer
- in the label field type a custom option (eg `{{name}}`)
- click on create option

*Expected Result*

The application works

*Current Result*

The application freeze

- [x] Not browser related

<details><summary> <b>Browser info</b> </summary>

(use this site: <a href="https://www.whatsmybrowser.org/">https://www.whatsmybrowser.org/</a> for non expert users)

| Browser Affected | Version |
|---|---|
|Internet Explorer| |
|Edge| |
|Chrome| |
|Firefox| |
|Safari| |
</details>

## Other useful information

It's probably due to this two use effect
https://github.com/geosolutions-it/MapStore2/blob/master/web/client/components/styleeditor/Fields.jsx#L230-L237

I would suggets to change this code and use a single useEffect with the stringify options dependency. It should not be an issue for preformance due the fact we usually have small arrays.

```js
useEffect(() => {
  setNewOptions(initOptions(options));
  // we should compare the previous and new options inside the dependencies
  // we could try the stringify for now because we have a usually small and constant array
  // this is to avoid infinite loop inside the style editor
}, [JSON.stringify(options)]);
```

**Comment by ElenaGallo** [comment]
@allyoucanmap there are some style problem:

- The **toc toolbar** and the **search layers bar** are moved up
![1](https://user-images.githubusercontent.com/56537133/121199415-33283780-c873-11eb-9048-cf4e959bafec.png)

- The color of the **family font** on visual editor is lighter
![2](https://user-images.githubusercontent.com/56537133/121200573-25bf7d00-c874-11eb-8edf-58fc34fb36bd.png)

- The **close button** in the save changes pop up and the **delete button** in delete pop up are moved up
![3](https://user-images.githubusercontent.com/56537133/121200603-29eb9a80-c874-11eb-90dd-95e4ce3e7ae8.png)
![4](https://user-images.githubusercontent.com/56537133/121200608-2b1cc780-c874-11eb-98f7-b551094d23ba.png)

**Comment by allyoucanmap** [comment]
@ElenaGallo I think this https://github.com/geosolutions-it/MapStore2/issues/6951#issuecomment-856803729 is duplicated. This issue of style should be fixed here #6946
